utils/angular-fetch.ts

```diff
@@ -1,37 +1,33 @@
-export default (angularHttpClient:any)=> (url: string, params: { headers: Record<string, string>, method: "GET" | "POST" | "PUT", body: string }) => {
-    const {headers, method, body} = params
+export default (angularHttpClient: any) => (url: string, params: { 
+    headers: Record<string, string>, 
+    method: "GET" | "POST" | "PUT", 
+    body?: string 
+}) => {
+    const { headers, method, body } = params;
+    const options = { headers, observe: 'response', responseType: 'text' };
+
+    const buildResponse = (response: any, ok: boolean) => {
+        const { status, headers, body, error, message } = response;
+        return {
+            status: status ?? (ok ? 200 : 500),
+            ok,
+            headers: { get: (name: string) => headers?.get?.(name) ?? null },
+            text: () => Promise.resolve(body ?? error ?? message ?? ''),
```

Comment:
An issue I can see here is that I think the body could be a text or a json, ultimately giving `[Object, object]` so could we test this case and if needed, stringify the body if `typeof body ?? error ?? message ?? '') !== string`?